plugin/trino-hive-hadoop2/src/test/java/io/trino/plugin/hive/s3select/TestHiveFileSystemS3SelectPushdownWithSplits.java

```diff
@@ -0,0 +1,127 @@
+import static io.trino.plugin.hive.HiveColumnHandle.ColumnType.REGULAR;
+import static io.trino.plugin.hive.HiveColumnHandle.createBaseColumn;
+import static io.trino.plugin.hive.HiveFileSystemTestUtils.filterTable;
+import static io.trino.plugin.hive.HiveFileSystemTestUtils.newSession;
+import static io.trino.plugin.hive.HiveType.HIVE_INT;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.testing.QueryAssertions.assertEqualsIgnoreOrder;
+
+public class TestHiveFileSystemS3SelectPushdownWithSplits
+{
+    private String host;
+    private int port;
+    private String databaseName;
+    private String awsAccessKey;
+    private String awsSecretKey;
+    private String writableBucket;
+    private String testDirectory;
+    private SchemaTableName tableWithSplits;
+
+    @Parameters({
+            "hive.hadoop2.metastoreHost",
+            "hive.hadoop2.metastorePort",
+            "hive.hadoop2.databaseName",
+            "hive.hadoop2.s3.awsAccessKey",
+            "hive.hadoop2.s3.awsSecretKey",
+            "hive.hadoop2.s3.writableBucket",
+            "hive.hadoop2.s3.testDirectory",
+    })
+    @BeforeClass
+    public void setup(String host, int port, String databaseName, String awsAccessKey, String awsSecretKey, String writableBucket, String testDirectory)
+    {
+        this.host = host;
+        this.port = port;
+        this.databaseName = databaseName;
+        this.awsAccessKey = awsAccessKey;
+        this.awsSecretKey = awsSecretKey;
+        this.writableBucket = writableBucket;
+        this.testDirectory = testDirectory;
+
+        tableWithSplits = new SchemaTableName(databaseName, "trino_s3select_test_csv_scan_range_pushdown");
+    }
+
+    @DataProvider(name = "testSplitSize")
+    public static Object[][] splitSizeParametersProvider()
+    {
+        return new Object[][] {{3, 2}, {50, 30}};
+    }
+
+    @Test(dataProvider = "testSplitSize")
+    public void testQueryPushdownWithSplitSize(int maxSplitSizeKB, int maxInitialSplitSizeKB)
+            throws Exception
+    {
+        S3SelectTestHelper s3SelectTestHelper = null;
+        try {
+            HiveConfig hiveConfig = new HiveConfig()
+                    .setS3SelectPushdownEnabled(true)
+                    .setMaxSplitSize(DataSize.of(maxSplitSizeKB, KILOBYTE))
+                    .setMaxInitialSplitSize(DataSize.of(maxInitialSplitSizeKB, KILOBYTE));
+            s3SelectTestHelper = new S3SelectTestHelper(
+                    host,
+                    port,
+                    databaseName,
+                    awsAccessKey,
+                    awsSecretKey,
+                    writableBucket,
+                    testDirectory,
+                    hiveConfig);
+            List<ColumnHandle> indexColumn = ImmutableList.of(
+                    createBaseColumn("index", 0, HIVE_INT, BIGINT, REGULAR, Optional.empty()));
+
+            assertEqualsIgnoreOrder(
+                    filterTable(
```

Comment:
I did a check via debug how many splits are collected at https://github.com/trinodb/trino/blob/98e5607fdb08005979bc9cefcd857339398970e1/plugin/trino-hive/src/test/java/io/trino/plugin/hive/HiveFileSystemTestUtils.java#L126 and got the following results for (maxSplitSize, maxInitialSplitSize):
- `(3,2)` -> `21` splits
-  `50,30` -> `3` splits  (the files are 13 KB each)

Again since this change is about controlling the number of the splits, I think we should check the output. In case of  `(3,2)` the number of splits could probably vary, so maybe using a function parameter  (in order to make something like `splitsCount between 15 and 25` - in java - would be appropriate).

Reply:
Updated

Reply:
Great. Thank you.